e2e/test/iothub/method/MethodE2ETests.cs

```diff
@@ -142,18 +143,22 @@ public async Task Method_ServiceInvokeDeviceMethodWithUnknownDeviceThrows()
             };
 
             // act
-            ErrorCode actualErrorCode = ErrorCode.InvalidErrorCode;
+            HttpStatusCode errorCode = HttpStatusCode.OK;
+            IotHubStatusCode iotHubStatusCode = IotHubStatusCode.Unknown;
             try
             {
                 // Invoke the direct method asynchronously and get the response from the simulated device.
                 await serviceClient.DirectMethods.InvokeAsync("SomeNonExistantDevice", methodInvocation);
             }
-            catch (DeviceNotFoundException ex)
+            catch (IotHubServiceException ex) when (ex.StatusCode is HttpStatusCode.NotFound)
```

Comment:
With "when", don't we need a catch all to cover other cases?

Reply:
No, we intentionally catch `IotHubServiceException` with `StatusCode` is `NotFound` here.  

That's the only expected exception thrown out here, otherwise we should let it throw out unexpected exceptions.